src/common/sha2/sha2_ossl.c

```diff
@@ -114,20 +120,21 @@ void OQS_SHA2_sha512_inc_init(OQS_SHA2_sha512_ctx *state) {
 	md = oqs_sha512();
 	OQS_EXIT_IF_NULLPTR(md);
 	mdctx = EVP_MD_CTX_new();
-	EVP_DigestInit_ex(mdctx, md, NULL);
+	OQS_EXIT_IF_NULLPTR(mdctx);
+	OQS_OPENSSL_GUARD(EVP_DigestInit_ex(mdctx, md, NULL));
 	state->ctx = mdctx;
 }
 
 void OQS_SHA2_sha512_inc_blocks(OQS_SHA2_sha512_ctx *state, const uint8_t *in, size_t inblocks) {
-	EVP_DigestUpdate((EVP_MD_CTX *) state->ctx, in, inblocks * SHA2_BLOCK_SIZE);
+	OQS_OPENSSL_GUARD(EVP_DigestUpdate((EVP_MD_CTX *) state->ctx, in, inblocks * 2 * SHA2_BLOCK_SIZE));
```

Comment:
We'll have to revert f57ed6db90b4886a95e8cd5a885bc2660bbdf134 on #1420 but that's fine.